src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/RecordMapper.java

```diff
@@ -40,14 +40,31 @@
     @Override
     public MtRecord apply(Record record) {
         final StreamRecord streamRecord = record.getDynamodb();
-        return new MtRecord()
-            .withAwsRegion(record.getAwsRegion())
-            .withEventID(record.getEventID())
-            .withEventName(record.getEventName())
-            .withEventSource(record.getEventSource())
-            .withEventVersion(record.getEventVersion())
-            .withUserIdentity(record.getUserIdentity())
-            .withContext(mtContext.getContext())
+        MtRecord ret = new MtRecord();
+        if (record.getAwsRegion() != null) {
```

Comment:
What's the reason for this change? I believe most of these fields usually are not null, so now we're calling each getter twice.

Reply:
so the reason for this is, we have tests creating mocked Record instances, so all these are nulled in said tests. 

Java has no problem letting these values be set to null, but when I converted to Kotlin, it set all the types to be Non-null, enforced at runtime. So we have three options: 
1) when translating to Kotlin, keep the lax nullable values
2) Update the test to go further with the mock and set default values (or use real Record instances)
3) enforce the non-null fields in MtRecord constructors and factories

This is option 3), but thinking about it, option 2) might be safer. Kind of kills the actual proper null handling of Kotlin though, if we let everything be nullable, sort of introducing tech debt to a solvable problem (that's much uglier/harder/prohibitively tedious to solve in Java)

Reply:
advice on which of the above 3 approaches are preffered would be appreciated. Again, option 3) has been merged in, but I can see objections to muddying prod code to deal with test reqs (converting mock `Record` objects to real `MtRecord` instances.

Reply:
If Kotlin is happy with what you have, then I am fine keeping it this way. Was just trying to understand the reason. If this is necessary, it may also be useful to add a comment about it. Also makes me wonder if this difference in null support could come up in other places.